### PR TITLE
Synchronized, Volatile and Blocking queue

### PR DESCRIPTION
**Pull request by Yanetta**

### CODE REVIEW

File: src/main/java/multuthreadingAndBlockingQueue/Account.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    public synchronized void depositDollars(long amount) {
+        if (amount < 0) {
+            throw new IllegalArgumentException("negative amount");
+        } else {
+            dollarBalance += amount;
+        }
+    }
+
+    public void withdrawDollars(long amount) {
+        synchronized (this) {
```
Comment (SerhiiPasko) [inline src/main/java/multuthreadingAndBlockingQueue/Account.java:40]: What this give you?

File: src/main/java/multuthreadingAndBlockingQueue/Account.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            if (amount < 0) {
+                throw new IllegalArgumentException("negative amount");
+            } else if (dollarBalance < amount) {
+                throw new IllegalArgumentException("not enough money");
+            }
+            dollarBalance -= amount;
+        }
+    }
+
+    public void depositHryvhya(long amount) {
```
Comment (SerhiiPasko) [inline src/main/java/multuthreadingAndBlockingQueue/Account.java:50]: I think, you wrote same logic for gruvna and dollar for show synchonised only?

File: src/main/java/multuthreadingAndBlockingQueue/Producer.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package multuthreadingAndBlockingQueue;
+
+import java.util.concurrent.BlockingQueue;
+import java.util.Random;
+
+public class Producer implements Runnable {
+    private BlockingQueue <Integer> bq;
+
+    public Producer(BlockingQueue <Integer> queue) {
+        this.bq = queue;
```
Comment (SerhiiPasko) [inline src/main/java/multuthreadingAndBlockingQueue/Producer.java]: What reason of "this" in this case?

File: src/main/java/multuthreadingAndBlockingQueue/Main.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        withdrawalThread2.start();
+
+        try {
+            Thread.sleep(4000);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+
+        System.out.println("BlockingQueue examples");
+        BlockingQueue<Integer> bq = new LinkedBlockingQueue<>(1);
```
Comment (SerhiiPasko) [inline src/main/java/multuthreadingAndBlockingQueue/Main.java]: For one element present special collections

**Review by SerhiiPasko — APPROVED** [review]
Good

**Review by SerhiiPasko — CHANGES_REQUESTED** [review]
Sorry, for previous result. I don't see synchronized block

File: src/main/java/multuthreadingAndBlockingQueue/Account.java (hunk 1 above)
Comment (Yanetta) [reply src/main/java/multuthreadingAndBlockingQueue/Account.java:40]: This synchronized block works like synchronized method. I have two threads which work with  withdrawDollars method. WithdrawDollars method changes dollarBalance variable on my single object "account".  I synchronized this method on this object . Thank to this only one thread will have a possibility to execute code in a synchronized block at certain moment of time.

File: src/main/java/multuthreadingAndBlockingQueue/Account.java (hunk 2 above)
Comment (Yanetta) [reply src/main/java/multuthreadingAndBlockingQueue/Account.java:50]: Methods for hryvnya are not synchronized, but work with volatile variable hryvnyaBalance;

File: src/main/java/multuthreadingAndBlockingQueue/Producer.java (hunk 3 above)
Comment (Yanetta) [reply src/main/java/multuthreadingAndBlockingQueue/Producer.java]: Just to emphasize that it is the variable of this object )). I will remove "this".

File: src/main/java/multuthreadingAndBlockingQueue/Main.java (hunk 4 above)
Comment (Yanetta) [reply src/main/java/multuthreadingAndBlockingQueue/Main.java]: OK, I'll make changes. I just wanted to examine how method take removes elements)

**Comment by Yanetta** [comment]
My synchronized block is inside withdrawDollars method, or it is not what you mean?

File: src/main/java/multuthreadingAndBlockingQueue/Main.java [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+        new Thread(consumer).start();
+        
+        InterestPayment interestPayment = new InterestPayment(account);
+        InterestInfo interestInfo = new InterestInfo(account);
+
+        new Thread(interestPayment).start();
+        new Thread(interestInfo).start();
+    }
+}
+class InterestPayment implements Runnable {
```
Comment (SerhiiPasko) [inline src/main/java/multuthreadingAndBlockingQueue/Main.java:97]: Why did you do same logic in main? SOLID, "S"

**Review by SerhiiPasko — APPROVED** [review]
Good. Read comments

**Pull request merged**